internal/api/fetcher/configapi.go

```diff
@@ -35,17 +42,47 @@ func (f *ConfigAPIFetcher) FetchAzureSEVSNPVersionList(ctx context.Context, atte
 
 // FetchAzureSEVSNPVersion fetches the version information from the config API.
 func (f *ConfigAPIFetcher) FetchAzureSEVSNPVersion(ctx context.Context, attestation configapi.AzureSEVSNPVersionGet) (configapi.AzureSEVSNPVersionGet, error) {
-	// TODO(elchead): follow-up PR for AB#3045 to check signature (sigstore.VerifySignature)
+	urlString, err := attestation.URL()
+	if err != nil {
+		return attestation, err
+	}
+	versionBytes, err := fetchBytesFromRawURL(ctx, urlString, f.httpc)
+	if err != nil {
+		return attestation, fmt.Errorf("fetch version %s: %w", attestation.Version, err)
+	}
+
+	signature, err := fetchBytesFromRawURL(ctx, fmt.Sprintf("%s.sig", urlString), f.httpc)
+	if err != nil {
+		return attestation, fmt.Errorf("fetch version %s signature: %w", attestation.Version, err)
+	}
+
+	err = sigstore.VerifySignature(versionBytes, signature, f.cosignPublicKey)
+	if err != nil {
+		return attestation, fmt.Errorf("verify version %s signature: %w", attestation.Version, err)
+	}
 	return fetch(ctx, f.httpc, attestation)
 }
 
+func fetchBytesFromRawURL(ctx context.Context, urlString string, client HTTPClient) ([]byte, error) {
+	url, err := url.Parse(urlString)
+	if err != nil {
+		return nil, fmt.Errorf("parse version url %s: %w", urlString, err)
+	}
+	bytes, err := FetchFromURL(ctx, client, url)
+	if err != nil {
+		return nil, err
+	}
+	return bytes, nil
```

Comment:
```suggestion
	return FetchFromURL(ctx, client, url)
```
If there is no additional error wrapping, you may just return the result of the function directly